mftf/2.3/reporting.md

```diff
@@ -0,0 +1,367 @@
+ Value was saved.
+ I magento cli "cache:clean config full_page"
+ Cleaned cache types:
+config
+full_page
+ I am on page "/admin/admin"
+ I wait for page load
+ I fill field "#username","admin"
+ I fill field "#login","123123q"
+ I click ".actions .action-primary"
+ I wait for page load 30
+ I close admin notification
+ I click "//li[@id='menu-magento-backend-stores']"
+ I wait for loading mask to disappear
+ I click "#nav li[data-ui-id='menu-magento-config-system-config']"
+ I wait for page load
+ I see current url matches "~\/admin\/system_config\/~"
+ I see "#something"
+ I save screenshot
+ FAIL
+
+ I magento cli "config:set admin/security/use_form_key 0"
+ Value was saved.
+ I magento cli "cache:clean config full_page"
+ Cleaned cache types:
+config
+full_page
+ I am on page "/admin/admin/auth/logout/"
+--------------------------------------------------------------------------------
+```
+{: .no-copy }
+
+The general test details and scenario has the same format as in the Passed test.
+The interesting part starts near the `FAIL` line.
+
+```terminal
+ I see "#something"
+ I save screenshot
+ FAIL
+```
+
+When a test step fails, the MFTF always saves a screenshot of the web page with failing state.
+And it always follows the failing test step, which is `I see "#something"` in our case.
+
+A screenshot of the fail goes at the `acceptance/tests/_output` directory in both PNG and HTML formats:
+
+- `Magento.AcceptanceTest._default.Backend.AdminMenuNavigationWithSecretKeysTestCest.AdminMenuNavigationWithSecretKeysTest.fail.html`
+- `Magento.AcceptanceTest._default.Backend.AdminMenuNavigationWithSecretKeysTestCest.AdminMenuNavigationWithSecretKeysTest.fail.png`
+
+The file name encodes:
+
+- `Magento` namespace
+- with the `AcceptanceTest` test type
+- generated as a part of the `_default` suite
+- defined at the `Magento_Backend` module
+- implemented in the `AdminMenuNavigationWithSecretKeysTestCest` PHP class
+- with the `AdminMenuNavigationWithSecretKeysTest` test name
+- and execution status `fail`
+
+Actions after `FAIL` are run as a part of the [`after`][] hook of the test.
+
+### Test result report
+
+After the MFTF completed test execution, it generates a general report about test results along with detailed information about each fail.
+
+```terminal
+--------------------------------------------------------------------------------
+DEPRECATION: Calling the "Symfony\Component\BrowserKit\Client::getInternalResponse()" method before the "request()" one is deprecated since Symfony 4.1 and will throw an exception in 5.0. /Users/.../magento2ce/vendor/symfony/browser-kit/Client.php:208
+
+Time: 52.43 seconds, Memory: 16.00MB
+
+There was 1 failure:
+---------
+```
+{: .no-copy }
+
+First you see warnings and deprecations.
+The `DEPRECATION` here is thrown by an MFTF dependency (Symfony) that is out of the scope for test writers and should be considered by MFTF contributors.
+In such cases, [report an issue][].
```

Comment:
If you encounter these types of errors,

Reply:
These are not errors in fact. 
Will change to: If you encounter this type of reporting,